Scripts/Runtime/Data/Collections/UnsafeTypedStream.cs

```diff
@@ -589,12 +586,32 @@ public int Count()
                 int count = 0;
                 for (int i = 0; i < m_BufferInfo->LaneCount; ++i)
                 {
-                    LaneInfo* lane = m_BufferInfo->LanesInfo + i;
+                    LaneInfo* lane = m_BufferInfo->LaneInfos + i;
                     count += lane->Count;
                 }
 
                 return count;
             }
+
+            public void ReadInto(ref NativeArray<T> array)
```

Comment:
- We should probably rename as `CopyTo` for consistency. `Read`, to me, implies stateful reading where the read head will be advanced to the end of the content delivered.
 - Might make sense to move this method up to `UnsafeTypedStream`

Reply:
> Might make sense to move this method up to UnsafeTypedStream

I don't want to move it up because the most common use case is have the `Reader` only in your job and not the full structure. You bring up a  good point though so I exposed the ability to do it from the full structure which drills down through a reader instance.

> We should probably rename as CopyTo for consistency. Read, to me, implies stateful reading where the read head will be advanced to the end of the content delivered.

I agree but this is stateful reading. The reader head will be moved to the end for that instance of the reader. I need to do this so that we can detect when we're at the end of a block and change to the next one.

Thoughts?

Reply:
Maybe I'm missing something but this method doesn't change the state of the reader does it?
Your LaneReaders move through a state but from what I understand they're aligned to the first element of the first block with each instance you get back from an `AsLaneReader()` call.

Calling `ReadInto()` doesn't change the state of your `Reader` instance. Meaning if you call `ReadInto()` on this instance and then try to perform other other read operations with this instance it behaves the same way. The ReadInto doesn't consider an existing reader state or affect the reader state. It just dumps all of the streams data into an array from the first element of the first lane to the last element of the last lane.